controllers/assignments.py

```diff
@@ -632,8 +632,9 @@ def chooseAssignment():
 def _get_lti_record(oauth_consumer_key):
     return db(db.lti_keys.consumer == oauth_consumer_key).select().first()
 
-def _get_course_practice_record(course_name):
-    return db(db.course_practice.course_name == course_name).select().first()
+def _get_course_practice_record(course_name, user_id):
```

Comment:
Do we know that both course_name and user_id will be not None ?  What if they are?

Reply:
@bnmnetp, I do not call this function if either of them is none. You can verify this by looking at all the places that I've invoked this function.